File: conversion_scripts/domain/backgrounds.py
"""Defines the domain of background definitions."""
from __future__ import annotations

from pydantic import Extra  # pylint: disable=unused-import
from pydantic import BaseModel, Field

from .books import Book


class Background(BaseModel, extra=Extra.forbid):
	"""Definition of a background."""

	name: str = Field(description="Background name.")
	source: Book = Field(description="Source book that defines the background.")

File: src/data/backgrounds.js
/**
 * Data definition for backgrounds.
 */

/**
 * Definition of a background.
 */
class Background {
    /**
     * @param {String} name_ Background name.
     * @param {Book} source Source book that defines the background.
     */
    constructor ({ name_, source }) {
        this.name_ = name_;
        this.source = source;
    }
};

export { Background };
